test/utils/castToJestMock.ts

```diff
@@ -1 +1,2 @@
-export const castToJestMock = (obj: any): jest.Mock => obj;
+export const castToJestMock = <T extends (...args: any[]) => any>(obj: T) =>
```

Comment:
We need to fix this line in [accordance to our ESLint rule](https://app.circleci.com/pipelines/github/algolia/instantsearch.js/5623/workflows/77f5063e-54c3-4d4f-84f9-986ffcb8db4a/jobs/25854/parallel-runs/0/steps/0-106).